### ISSUE TITLE
Remove two TODO comments from the help text

### ISSUE THREAD

**Issue by jostephd** [wesnoth/wesnoth#4280]
Remove the two TODO comments added in #4217. I need some input about _how_ to fix this, see https://github.com/wesnoth/wesnoth/pull/4217#issuecomment-524574853

This issue is just about those two comments. For other todo comments in the help see #2964

- [x] Fix `At all difficulties except “Beginner” (TODO: and “Easy”?),`
- [x] Fix `TODO: link to a forum thread`
- [x] Fix `At least for the 1.15.x series before 1.16.0` (see next comment)
- [x] Does `In the mainline campaigns, on hard` mean "on each campaign's hardest difficulty"? Even if the hardest difficulty is "Nightmare"?

**Comment by stevecotton** [comment]
Thanks for logging this, it's probably a good ticket to fix the save-loading page related "refer to the forums" line:

https://github.com/wesnoth/wesnoth/blob/c4ffade49aa4efc2a410e031539a791c005c3b30/data/core/help.cfg#L289-L294

**Comment by jostephd** [comment]
Agreed, added a checklist to the first post. Also marked as blocker, not in any technical sense but because it's the sort of thing that should be fixed before a final release.

**Comment by jostephd** [comment]
This is a string freeze blocker. Could someone take this issue on and fix it? That is, either draft a PR, or start a discussion first to figure out what the PR should say?

**Comment by stevecotton** [comment]
We've already had a discussion without reaching consensus.
https://forums.wesnoth.org/viewtopic.php?f=3&t=18640

**Comment by jostephd** [comment]
I remember that, but even though we didn't reach consensus, we still need to fix the todos before string freeze, one way or the other. For example, maybe we can agree that "winnable without foreknowledge" is always true when difficulty is at most "Normal" and level is at most "Intermediate", and leave the expectation of higher difficulties/levels unspecified. Or maybe we'd just revise those bullets into recommended rules of thumb for creating UMC campaigns. Or something.

**Comment by stevecotton** [comment]
@slavrenyuk has brought up a similar discussion on Discord https://www.wesnoth.org/irclogs/2019/11/%23wesnoth-dev.2019-11-06.log

**Comment by CelticMinstrel** [comment]
Suggestion: Remove the TODO notes from the string and add them instead as a comment near the string. That way, at least they don't block the string freeze.

**Comment by stevecotton** [comment]
If they're incorrect information, then I'd consider them blockers, and incorrect-but-looks-correct is worse than incorrect-and-obviously-so.

**Comment by slavrenyuk** [comment]
@Pentarctagon what do you think about this https://github.com/wesnoth/wesnoth/blob/master/data/core/help.cfg#L336?
I would rephrase that TODO as 
>in the game help, link a forum thread where new players can post their mainline singleplayer replays/questions and get review/corrections

Perhaps it would be better to have a separate topic for each mainline campaign?

**Comment by slavrenyuk** [comment]
Regarding 
> At all difficulties except “Beginner” (TODO: and “Easy”?), campaigns assume that you will have some carryover gold and some advanced units on your recall list.

I think `and “Easy”` should be removed. My friend recently started to play Wesnoth, she (more or less) easily finished several campaigns on "Beginner" difficulty, but she wasn't able to finish Liberty on "Easy", yet she had some carryover gold and veterans.

**Comment by Pentarctagon** [comment]
@slavrenyuk The Strategies & Tips forum is already for that, in part, so I don't know if a specific thread is really needed.  I'd also be a bit worried about new players' impressions if we provide a specific thread, they post their replay there, and then nobody replies.  Creating a specific thread for it feels like it creates more of an expectation that they'll get a reply when that might not be the case (thinking of the Android support thread here, for example).

**Comment by slavrenyuk** [comment]
I agree with @Pentarctagon

**Comment by stevecotton** [comment]
Maybe
* a link in the help to the strategies & tips forum
* a per-campaign link to the feedback thread, with the note that it may contain useful tips and replays
    - for mainline campaigns, to the index thread

**Comment by Pentarctagon** [comment]
Sounds good to me, as long as having links to all 18 campaign index threads doesn't end up really cluttered.

**Comment by Pentarctagon** [comment]
What's the status of this?

**Comment by CelticMinstrel** [comment]
If the todo comments are still in the help text, then I would say just remove them. You could add them as actual WML comments elsewhere, if they need to be preserved.

**Comment by stevecotton** [comment]
Given that they're rebalancing the campaigns, maybe @nemaara and @Hejnewar would be able to give an answer.

@CelticMinstrel - for this, I think incorrect documentation would be worse than the visible todo comments.

**Comment by nemaara** [comment]
Personally I think having that as a help topic feels a little bit janky, it would feel much better as a steam thread or forum post imo. I think an even better solution would be to have the campaigns feel more balanced overall even for newer players, which I'll hopefully work on for 1.16 (no guarantees on doing all the campaigns, but at least some of them).

Soliton said this before too, but it's something I totally agree with - on easy mode, campaign difficulty generally should be easy enough that most people can play to explore the story without worrying too much about losing (yes, there are some exceptions like UtBS and NR where I would expect the player to be at least moderately skilled to pass on easy). 

Regarding those comments in particular, I don't necessarily make assumptions about recalls or or carryover gold because every campaign is different. It's really hard to make catchall rules for all the campaigns given how disparate they can be in design, so I'm not sure it's a good idea to put anything down like that.

**Comment by CelticMinstrel** [comment]
> for this, I think incorrect documentation would be worse than the visible todo comments.

Then remove the incorrect documentation; if it's partially correct and you need to preserve it, then it can be moved to the comments. There should not be TODO comments in translatable strings under any circumstances.

**Comment by stevecotton** [comment]
> on easy mode, campaign difficulty generally should be easy enough that most people can play to explore the story without worrying too much about losing (yes, there are some exceptions like UtBS and NR where I would expect the player to be at least moderately skilled to pass on easy).

One of the inspirations for this help topic was a study of some completely new players' first attempts at Wesnoth which found that the game is much harder than we think. My memory's hazy on this, but IIRC  @sigurdfdragon did the study, and 60% of players died on tutorial S02.

**Comment by nemaara** [comment]
Ya I totally agree with that. I actually asked EarthCake to draw a new Tutorial S2 map for that reason and generally made TSG, Liberty, and DiD easier than they were before. I think overall the other campaigns could use some toning down at least on easy mode.

Still, I think this is an issue with balancing, I feel like it's really meh to stick that topic in the game itself since it sort of gives off the impression that we assume the players are going to lose.

**Comment by stevecotton** [comment]
> Still, I think this is an issue with balancing, I feel like it's really meh to stick that topic in the game itself since it sort of gives off the impression that we assume the players are going to lose.

While they're learning the mechanics and basic strategies of the game, I think we should expect them to lose sometimes, even on easy. Some of the basic maths of the game is counter-intuitive, for example that 3 x 70% doesn't guarantee a hit, or that a 1% chance for a hero to die is a high probability.

**Comment by nemaara** [comment]
I don't really agree with that. If new players go play AToTB or TSG on the easiest mode, I don't think they should lose on most scenarios. Even something like Liberty or the early scenarios of HttT should be pretty accessible to new players. I don't see a reason to gatekeep the mainline campaigns based on difficulty, players (yes even brand new players never touched strategy games before) should be able to have a fun time regardless of their skill level. Of course there are some scenarios that may take a couple tries, such as Siege of Elensefar or Choice in the Fog, but I believe that this is natural for most people.

The basic maths of the game being counterintuitive are a gameplay flaw to me if we don't balance around them properly. What I mean is that there's no reason to not offer a better learning curve for new players to learn that on their own rather than have us explicitly spell it out for them in the game. I don't think explicitly putting in something that suggests that we expect players to lose is a good idea.

Elaborating on the above - 

1. I don't think we should expect players to lose very often for scenarios of the appropriate skill level. There are some scenarios where we might expect them to lose, these would be challenge scenarios like Siege of Elensefar. We should try to balance them carefully so that they're difficult and may require a couple tries, but allow players to feel accomplished after beating them. There should not be too many of these in a single campaign, and they should be spread out (yes, there can be exceptions like for NR and UtBS which are meant to be challenge campaigns).

2. I personally would not purposely try to design a game where most people would get frustrated enough to start scrolling through pages of strategies just to pass some of the easier campaigns. The presence of that topic _in the game itself_ would read to me like we expect people to get frustrated, even if that's not its intention. If it has to stay, I would suggest rewording its title and changing a lot of its text. Some of it is really vague and not helpful, while other parts hint at what I think is not good game design.

3. Repeating because this is important. I don't think we need to gatekeep the game to only skilled players. If people are new to or bad at strategy games, fine. They should still be able to have fun. I don't think losing repeatedly (and in fact the game designer expecting you to lose repeatedly) is fun. 

4. Of course the game has a lot of strategic depth, that's why we have expert level campaigns to reflect that. Once a player tries to play those and find that it's too difficult for them, I suspect that they're invested enough to start looking for strategies and tips on their own, which is something that I would not mind having in the game under an advanced strategies section. I think this is different in concept than a "why did I lose" topic.

**Comment by stevecotton** [comment]
@nemaara, please would you explain which parts of the text you're responding to?

> I don't think they should lose on most scenarios

Where did either the text or my comment say "most" or "very often"? The text is about losing some of the time, not most.

> I don't see a reason to gatekeep the mainline campaigns based on difficulty

Where does this concept of gatekeeping come from? Where in the text that you're reviewing is it implied?

> Of course there are some scenarios that may take a couple tries, such as Siege of Elensefar or Choice in the Fog, but I believe that this is natural for most people.

This sounds like you're expecting them to lose too. However, the suggestion that retrying the scenario is enough is one of the misconceptions that the text is addressing - SoE is not a scenario in isolation, it's a part of a campaign. How would you design the game to enable most players who barely scraped victory in the preceding scenarios to win SoE, while still giving a positive reward for players who did better in the preceding scenarios?

**Comment by Pentarctagon** [comment]
Throwing my own opinion in here: I think it's a good goal to have that all mainline campaigns should be beatable my everyone on Easy and rebalancing existing campaigns with that goal in mind is definitely worthwhile, however there will also always be people who misjudge their own skill and so choose a harder difficulty than they should, are simply not good at tactical games, try to force strategies from other games to work that just don't (ie: treating standard Wesnoth campaign leaders like RPG characters that get extremely powerful), get really bad luck after exposing their leader more than necessary, etc.  Short of introducing a "Story Mode" type of difficulty, and maybe even then, there will always end up being some number of users who manage to lose even the easiest scenarios/campaigns.

So I don't think there's anything fundamentally flawed about adding text to point those users in the direction of information that might help them get past the scenario they're having a problem with.

**Comment by nemaara** [comment]
Fine, just leave it then.

**Comment by stevecotton** [comment]
@Pentarctagon, please would someone review the associated PRs (#5076 and #5077)?

**Comment by Pentarctagon** [comment]
Help text would be something reviewed by vultraz normally, but he's mentioned he doesn't have a lot of time at the moment, so I'll go ahead and do it.

**Comment by CelticMinstrel** [comment]
So, this can be closed now, right?

**Comment by stevecotton** [comment]
Finished, it was waiting for #5083 to be merged and the update to the changelog and release notes.

#5083 is showing as closed rather than merged, this is because I pushed it to master along with the changelog update. It is merged.
